modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/tree/BplusTree.java

```diff
@@ -4550,10 +4551,10 @@ public long pollFreePage() {
                 return 0L;
             }
 
-            if (freePages.getClass() == IgniteLongList.class) {
-                IgniteLongList list = ((IgniteLongList) freePages);
+            if (freePages.getClass() == LongArrayList.class) {
```

Comment:
why not `instanceOf`? Also, maybe a queue is more suitable here?

Reply:
I think reference equality is used here for better performance, I need to think about the queue.

Reply:
I doubt that the performance difference is even measurable in this case

Reply:
I fixed it, hope it's better.